pennylane/transforms/optimization/cancel_inverses.py

```diff
@@ -64,6 +63,45 @@ def _are_inverses(op1, op2):
     return False
 
 
+def _can_cancel_ops(op1, op2):
+    """Checks if two operators can be cancelled
+
+    Args:
+        op1 (~.Operator)
+        op2 (~.Operator)
+
+    Returns:
+        Bool
+    """
+    if _are_inverses(op1, op2):
+        # Make sure that if one of the ops is Adjoint, it is always op2 by swapping
+        # the ops if op1 is Adjoint
+        if isinstance(op1, Adjoint):
+            op1, op2 = op2, op1
+
+        # If the wires are the same, then we can safely cancel both
+        if op1.wires == op2.wires:
+            return True
```

Comment:
If they are inverses, then there wires are the same?

Reply:
So I've updated the `_are_inverses` function to check if two operators are inverses independent of their wires. This was already the case for _most_ operators, but when checking `Adjoint` against another op, wires were also being compared.

Wire related validation should happen later because we can cancel if the wires are the same, but in some cases we can also cancel if the _set_ of wires/control wires is the same due to wire/control wire symmetry.